app/models/user.rb

```diff
@@ -24,6 +24,24 @@ class User < ActiveRecord::Base
   alias_method :role?, :roles?
   scope :hosts, -> { with_roles :host }
 
+  def name(format = nil)
+    names = [given_name, nickname, family_name]
+    case format
+    when :fullest
+      names[1] = "\"#{names[1]}\""
+    when :full
+      names.delete_at(1)
+    when nil
+      names = [names[1]]
+    end
+
+    names.reject(&:nil?).join(' ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.